src/swap/components/SwapProvider.tsx

```diff
@@ -118,30 +123,42 @@ export function SwapProvider({
   }, [hasHandledSuccess, lifeCycleStatus.statusName, resetInputs]);
 
   useEffect(() => {
-    const maxSlippage = lifeCycleStatus.statusData.maxSlippage;
     // Reset status to init after success has been handled
     if (lifeCycleStatus.statusName === 'success' && hasHandledSuccess) {
-      setLifeCycleStatus({
+      updateLifeCycleStatus({
         statusName: 'init',
         statusData: {
-          isMissingRequiredField:
-            lifeCycleStatus.statusData.isMissingRequiredField,
-          maxSlippage,
+          isMissingRequiredField: true,
+          maxSlippage: config.maxSlippage,
         },
       });
     }
   }, [
+    config.maxSlippage,
     hasHandledSuccess,
-    lifeCycleStatus.statusData,
     lifeCycleStatus.statusName,
+    updateLifeCycleStatus,
   ]);
 
   const handleToggle = useCallback(() => {
     from.setAmount(to.amount);
     to.setAmount(from.amount);
     from.setToken(to.token);
     to.setToken(from.token);
-  }, [from, to]);
+
+    updateLifeCycleStatus({
+      statusName: 'amountChange',
+      statusData: {
+        amountFrom: from.amount,
+        amountTo: to.amount,
+        tokenFrom: from.token,
+        tokenTo: to.token,
+        // token is missing
+        isMissingRequiredField:
+          !from.token || !to.token || !from.amount || !to.amount,
+      },
+    });
+  }, [from, to, updateLifeCycleStatus]);
```

Comment:
It seemed like we should be triggering an amountChange on toggle since from and to switch, thoughts?

Reply:
You right! Great point